cargo-dylint/tests/ci.rs

```diff
@@ -713,6 +713,70 @@ fn update() {
     });
 }
 
+#[test]
+fn lint_script_test() {
+    let mut restriction_libs = Vec::new();
+    for entry in read_dir("examples/restriction").unwrap() {
+        let entry = entry.unwrap();
+        if entry.path().is_dir() {
+            let lib_name = entry.file_name().into_string().unwrap();
+            // Exclude overscoped_allow as in the script
+            if lib_name != "overscoped_allow" && lib_name != ".cargo" {
+                restriction_libs.push(format!("--lib {lib_name}"));
+            }
+        }
+    }
+    let restrictions_as_flags = restriction_libs.join(" ");
+
+    // Initialize dirs_to_lint with static paths from an array
+    let mut dirs_to_lint: Vec<PathBuf> = [
+        PathBuf::from("."),
+        PathBuf::from("driver"),
+        PathBuf::from("utils/linting"),
+        PathBuf::from("examples/general"),
+        PathBuf::from("examples/supplementary"),
+        PathBuf::from("examples/restriction"),
+        PathBuf::from("examples/testing/clippy"),
+    ]
+    .to_vec();
+
+    // Iterate over experimental directories and push directly to dirs_to_lint
+    for entry in read_dir("examples/experimental").unwrap() {
+        let entry = entry.unwrap();
+        let path = entry.path(); // path is already a PathBuf
+        if path.is_dir() && path.file_name().unwrap() != ".cargo" {
+            dirs_to_lint.push(path);
+        }
+    }
+
+    let base_flags =
+        format!("--lib general --lib supplementary {restrictions_as_flags} --lib clippy");
+
+    eprintln!("Setting DYLINT_RUSTFLAGS='-D warnings' for each lint command");
+
+    for dir_path in &dirs_to_lint {
+        eprintln!("Linting in directory: {dir_path:?}");
+
+        let mut cmd = Command::cargo_bin("cargo-dylint")
+            .expect("Failed to find cargo-dylint binary. Ensure it is built (e.g. `cargo build -p cargo-dylint`) and in PATH or target/debug.");
```

Comment:
```suggestion
            .expect("Failed to find cargo-dylint binary");
```
Integration tests are supposed to run after binaries are built, so a user should not need to build `cargo-dylint` manually.